pkg/controller/account/account_controller.go

```diff
@@ -165,25 +163,50 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 	}
 
-	// We expect this secret to exist in the same namespace Account CR's are created
-	awsSetupClient, err := awsclient.GetAWSClient(r.Client, awsclient.NewAwsClientInput{
-		SecretName: utils.AwsSecretName,
-		NameSpace:  awsv1alpha1.AccountCrNamespace,
-		AwsRegion:  "us-east-1",
-	})
+	// If account CR is generated by accountpool_controller, there will be no ClaimLink available. If
+	// this field is empty set region to us-east-1 as AWS China only supports CCS/BYOC.
+	var awsRegion string
+	accountClaim := &awsv1alpha1.AccountClaim{}
+	if currentAcctInstance.Spec.ClaimLink == "" {
+		// Set region to us-east-1 if ClaimLink is not empty.
+		awsRegion = awsv1alpha1.AwsPlatformConfigGlobal.ClientInput.AwsRegion
+	}
+
+	// Return associated accountClaim.
+	accountClaim, err = r.ReturnAccountClaim(currentAcctInstance)
+	if err != nil {
+		reqLogger.Error(err, "Failed to return account from accountClaim")
+		return reconcile.Result{}, err
+	}
+
+	// Get region AWS Region to determine AWS Global or China platform.
+	awsRegion = accountClaim.Spec.Aws.Regions[0].Name
+
+	// Based on the awsRegion bool, import either the Global or China configuration.
+	AwsPlatformConfig := awsv1alpha1.AwsPlatformConfig{}
+
+	if !checkAwsProviderIsGlobal(awsRegion) {
+		AwsPlatformConfig = awsv1alpha1.AwsPlatformConfigChina
+	} else {
+		AwsPlatformConfig = awsv1alpha1.AwsPlatformConfigGlobal
+	}
+	reqLogger.Info(fmt.Sprintf("AWS Platform configured region: %s", AwsPlatformConfig.ClientInput.AwsRegion))
+
+	// Pass awsPlatformConfig to the setup client to create platform appropriate client.
+	awsSetupClient, err := awsclient.GetAWSClient(r.Client, AwsPlatformConfig.ClientInput)
```

Comment:
@dofinn Instead of taking the `AwsPlatformConfig` you could create a new function on the awsClient type that would take the AccountClaim and return a client that connects to the correct region based on the `spec.aws.regions`. You would need two functions something like a `GetAwsClientFromClaim` and a `GetAwsBYOCClientFromClaim` these respectively would return the correct client, the first returning one using the credentials from the root (payer) account and the latter returning a client from the credentials in referenced in the AccountClaim. With that we would not need to pass around the `AwsPlatformConfig` struct as the client is created correctly when its instantiated.

Reply:
It doesn't seem a new function has been created in the awsclient interface, receiving the accountClaim (which contains the region) and returning the client awsClient. Any reason for that?